lua/cmp_tabnine/source.lua

```diff
@@ -194,15 +207,40 @@ Source._on_stdout = function(_, data, _)
 			else
 				local results = response.results
 				old_prefix = response.old_prefix
+
 				if results ~= nil then
 					for _, result in ipairs(results) do
+						local newText = result.new_prefix .. result.new_suffix
+
+						local old_suffix = result.old_suffix
+						if string.sub(old_suffix, -1) == "\n" then
+							old_suffix = string.sub(old_suffix, 1, -2)
+						end
+
 						local item = {
-							label = result.new_prefix;
-							filterText = result.new_prefix;
-							insertText = result.new_prefix;
+							label = newText;
+							filterText = newText;
 							data = result;
-							sortText = result.new_prefix;
+							textEdit = {
+								range = {
+									start = { line = cursor.line, character = cursor.col - #old_prefix - 1 },
+									['end'] = {  line = cursor.line, character = cursor.col  + #old_suffix - 1 };
+								};
+								newText = result.new_prefix;
```

Comment:
why are u not using `newText` here?
this means that you should also include `new_suffix` in the edit.
Im not sure I completely understand the semantics of `textEdit`.

Reply:
Yes, please do.